dev/TreeView/ViewModel.h

```diff
@@ -83,10 +87,14 @@ class ViewModel :
     std::vector<winrt::event_token> m_IsExpandedChangedEventTokenVector;
     winrt::event_token m_rootNodeChildrenChangedEventToken;
     winrt::weak_ref<winrt::TreeViewList> m_TreeViewList{ nullptr };
+    winrt::weak_ref<winrt::TreeView> m_TreeView{ nullptr };
     tracker_ref<winrt::TreeViewNode> m_originNode{ this };
     bool m_isContentMode{ false };
     tracker_ref<winrt::IVector<winrt::IInspectable>> m_selectedItems{ this };
+    tracker_ref<winrt::IVector<winrt::IInspectable>> m_addedSelectedItems{ this };
+    tracker_ref<winrt::IVector<winrt::IInspectable>> m_removedSelectedItems{ this };
```

Comment:
I don't think these need to be tracker_ref variables. You need to use tracker_ref when you are holding a pointer to an object you don't own the lifetime of, in this case you own the vector. What you don't own are the objects you are putting into the vector, I think in this case this should be a vector of weak refs to objects.

Reply:
@ranjeshj to double check.  Maybe you need to ensure the removed items are kept alive and then you could use tracker ref's instead of weak_ref's

---
In reply to: [391306809](https://github.com/microsoft/microsoft-ui-xaml/pull/2063#discussion_r391306809) [](ancestors = 391306809)

Reply:
@ranjeshj what do you think about this ^ :-) ?

Reply:
Yes, I think these can be just a vector with weak refs except for the removed items.

Reply:
@ranjeshj you mean use 2 vectors (no `tracker_ref`) + strong ref for added items + weak ref for removed items?

Could you please explain why the weak ref would be needed only for removed items?

Reply:
> @ranjeshj you mean use 2 vectors (no `tracker_ref`) + strong ref for added items + weak ref for removed items?
> 
> Could you please explain why the weak ref would be needed only for removed items?
Its the other way. You can use a weak ref for added items (because the collection is already holding a reference). For removed items, you want to hold a strong ref since the collection is not holding it anymore and it might get destroyed if no one else has a reference to it.

Reply:
Got it, thanks for clarifying. 

What is the way to then retrieve items from the weak ref?

![image](https://user-images.githubusercontent.com/2237018/77087683-af7ef280-6a03-11ea-801f-f57d2f69123c.png)

Is there a better way other than creating a new vector, iterating over the weak refs and call `get()`

Reply:
Either that or update the SelectionChanged event handler to have a vector of weak_refs